src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,60 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+	
+	public static final String MESSAGE_CONTACT_CONSTRAINTS ="Person contact can be any form";
+    public static final String CONTACT_VALIDATION_REGEX = ".+";
+	public final String value;
+	private boolean isPrivate;
+
+	/**
+     * Validates given email.
+     *
+     * @throws IllegalValueException if given contact string is invalid.
+     */
+	public Contact(String contact, boolean isPrivate) throws IllegalValueException{
+		 this.isPrivate = isPrivate;
+	     String trimmedContact = contact.trim();
+	     checkValidation(trimmedContact);
+	     this.value=trimmedContact;
+	}
+	
+	/**
+     * Checks if a given string is a valid person contact.
+     * 
+     * @throws IllegalValueException if given contact string is invalid.
+     */
+	public void checkValidation(String trimmedContact) throws IllegalValueException {
+		if (!isValidContact(trimmedContact)) {
+	            throw new IllegalValueException(MESSAGE_CONTACT_CONSTRAINTS);
+	        }
+	}
+
+	/**
+     * Checks if a given string is a valid person contact.
+     */
+	public boolean isValidContact(String trimmedContact) {
+		return trimmedContact.matches(CONTACT_VALIDATION_REGEX);
+	}
+	
+    public String toString() {
+        return value;
+    }
+	
+	public boolean isPrivate() {
+        return isPrivate;
+    }
+    
+    public int hashCode() {
+        return value.hashCode();
+    }
+    
+    public boolean equals(Object other) {
```

Comment:
Beware of the issue we discussed in tutorial, i.e. two subclass objects with same value will be deemed equal.